cryptolib/signer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "errors"
+
+// Signer contains methods to create a signed Attestation.
+type Signer interface {
+	CreateAttestation(privateKey, payload []byte, publicKeyID string, alg SigningAlgorithm, keyMode KeyMode) (*Attestation, error)
```

Comment:
I think `privateKey` should be held in a config-like struct in the signer, instead of passing it as a parameter. The same should also apply to other metadata that are likely to remain constant for a signer instance, e.g., `SigningAlgorithm` or `keyMode`, `publicKeyID`.

Reply:
This would make sense to me if a user would want to create multiple attestations using the same private key.

For example, I can imagine it would be useful if the signer tool will be used to sign containers "in bulk" (e.g. a vulnerability signer processing a queue of container images). But I am not sure that this matches our current/near-future use case. Since you are working on the signer, could you provide more context on this?